explainability/explainability-core/src/test/java/org/kie/kogito/explainability/local/lime/optim/RecordingLimeExplainerTest.java

```
@@ -0,0 +1,99 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.kogito.explainability.local.lime.optim;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.jupiter.api.Test;
+import org.kie.kogito.explainability.Config;
+import org.kie.kogito.explainability.model.Prediction;
+import org.kie.kogito.explainability.model.PredictionProvider;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+class RecordingLimeExplainerTest {
+
+    @Test
+    void testRecordedPredictions() {
+        RecordingLimeExplainer recordingLimeExplainer = new RecordingLimeExplainer(10);
+        List<Prediction> allPredictions = new ArrayList<>();
+        PredictionProvider model = mock(PredictionProvider.class);
+        for (int i = 0; i < 15; i++) {
+            Prediction prediction = mock(Prediction.class);
+            allPredictions.add(prediction);
+            try {
+                recordingLimeExplainer.explainAsync(prediction, model).get(Config.DEFAULT_ASYNC_TIMEOUT,
+                        Config.DEFAULT_ASYNC_TIMEUNIT);
+            } catch (Exception e) {
+                // ignored for the sake of the test
```

Can you describe more please why is this ignored?

---

the exception is ignored because the test only cares about recording the explanation request; in fact both the `Prediction` and the `PredictionProvider` are mocked and therefore no explanation will be generated (hence the exception handling).